src/Features/Core/Portable/AddParameter/IAddParameterService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.AddParameter
+{
+    internal interface IAddParameterService
+    {
+        /// <summary>
+        /// Checks if there are indications that there might be more than one declarations that need to be fixed.
+        /// The check does not look-up if there are other declarations (this is done later in the CodeAction).
+        /// </summary>
+        bool HasCascadingDeclarations(IMethodSymbol method);
+
+        /// <summary>
+        /// Adds a parameter to a method.
+        /// </summary>
+        /// <param name="invocationDocument"></param>
```

Comment:
Empty descriptions can be omitted. The only interesting thing here is the `newParameterIndex` one.